### PR TITLE
Skip performOnPrimary step when executing PublishCheckpoint.

### PR DESCRIPTION
**Pull request by mch2**
### Description
This change fixes broken SegmentReplicationRelocationITs and unmutes them.  The cause of the flaky tests is WAIT_UNTIL requests blocking relocation of primary shards because replicas have not refreshed.  The issue is that primary shards cannot publish their checkpoints because PublishCheckpointAction is a ReplicationOperation, that first hits the primary and acquires a permit.  Given we are blocking operations, the permit is delayed and the request never succeeds.  This fixes the issue by ensuring PublishCheckpoint requests are not sent to the primary, and only to replicas within the replication group directly.

### Issues Resolved
closes https://github.com/opensearch-project/OpenSearch/issues/6065

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationRelocationIT.java [hunk 1]
```diff
@@ -43,22 +42,13 @@ public class SegmentReplicationRelocationIT extends SegmentReplicationBaseIT {
...
-        prepareCreate(
-            INDEX_NAME,
-            Settings.builder()
-                .put("index.number_of_shards", 1)
-                .put(IndexModule.INDEX_QUERY_CACHE_ENABLED_SETTING.getKey(), false)
-                .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
-                .put("index.number_of_replicas", replicaCount)
-                .put("index.refresh_interval", -1)
-        ).get();
+        prepareCreate(INDEX_NAME, Settings.builder().put(SETTING_NUMBER_OF_REPLICAS, replicaCount)).get();
```
Comment (dreamer-89) [inline server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationRelocationIT.java:45]: 👍

File: server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationRelocationIT.java [hunk 2]
```diff
@@ -43,22 +42,13 @@ public class SegmentReplicationRelocationIT extends SegmentReplicationBaseIT {
...
     }
 
     /**
      * This test verifies happy path when primary shard is relocated newly added node (target) in the cluster. Before
      * relocation and after relocation documents are indexed and documents are verified
      */
-    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/5669")
```
Comment (dreamer-89) [inline server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationRelocationIT.java:61]: nit: For completenes, do we need relocation tests for `IMMEDIATE` refresh policy ?

File: server/src/main/java/org/opensearch/indices/replication/checkpoint/PublishCheckpointAction.java [hunk 3]
```diff
@@ -146,12 +146,11 @@ public void handleResponse(ReplicationResponse response) {
                                 timer.time()
                             )
                         );
-                        task.setPhase("finished");
```
Comment (dreamer-89) [inline server/src/main/java/org/opensearch/indices/replication/checkpoint/PublishCheckpointAction.java:149]: Why this `setPhase` call is removed ? If it is not useful, let's remove same during failure as well (on line 156)

Comment (mch2) [reply server/src/main/java/org/opensearch/indices/replication/checkpoint/PublishCheckpointAction.java]: Not intentional, thanks for catching this, it is useful in tracking state in the `ReplicationTask`.

File: server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationRelocationIT.java (hunk 2 above)
Comment (mch2) [reply server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationRelocationIT.java:61]: Yes good idea, I've added randomness to select a refresh policy in all of the tests in this class other than the explicit wait_until test.

**Pull request merged**